R/IDateTime.R

```diff
@@ -5,9 +5,17 @@
 
 as.IDate <- function(x, ...) UseMethod("as.IDate")
 
-as.IDate.default <- function(x, ..., tz = attr(x, "tzone")) {
+i.IDate = NULL
+as.IDate.default <- function(x, ..., tz = attr(x, "tzone"), use_merge = 'auto') {
   if (is.null(tz)) tz = "UTC"
-  as.IDate(as.Date(x, tz = tz, ...))
+  if (isTRUE(use_merge) || (use_merge == 'auto' && length(x) >= 1000)) {
+    DT = data.table(x)
+    # shut off use_merge to prevent recursion
+    unique(DT)[ , 'IDate' := as.IDate(x, tz = tz, ..., use_merge = FALSE)
+                ][DT, on = 'x', IDate]
```

Comment:
You can use string or SDcols to avoid having to create dummy IDate variable

Reply:
yes, now that the order is switched (I don't think `'i.IDate'` would have worked?)